changelog/10839.deprecation.rst

```diff
@@ -0,0 +1 @@
+Synchronous tests that request an asynchronous fixture with ``autouse=True`` will now give a DeprecationWarning.
```

Comment:
We should add a new entry to `deprecations.rst`, explaining the rationale and how to fix the code, and add a link to that seciton here.